docs: add `Provider` doc section for `cacheUtxo` (#2598)
FuelLabs/fuels-ts: apps/docs/src/guide/provider/provider-options.md

_Note: If defined, `requestMiddleware`, `timeout` and `retryOptions` are applied to this custom `fetch` function as well._

<<< @/../../docs-snippets/src/guide/provider/provider.test.ts#options-fetch{ts:line-numbers}

### `cacheUtxo`

When using the SDK, it may be necessary to submit multiple transactions from the same account in a short period. In such cases, the SDK creates and funds these transactions, then submits them to the node.

However, if a second transaction is created before the first one is processed, there is a chance of using the same UTXO(s) for both transactions. This happens because the UTXO(s) used in the first transaction are still unspent until the transaction is fully processed.

If the second transaction attempts to use the same UTXO(s) that the first transaction has already spent, it will result in the following error:

```console
Transaction is not inserted. UTXO does not exist: 0xf5...
```

> This error indicates that the UTXO(s) used by the second transaction no longer exist, as they were already spent by the first transaction.

Reviewer: ```suggestion
This error indicates that the UTXO(s) used by the second transaction no longer exists, as it was already spent by the first transaction
```

Author: https://github.com/FuelLabs/fuels-ts/pull/2598/commits/0941eaa790b3355ab8aa0a6911fdb6c8da051b6b

> To prevent this issue, you can use the `cacheUtxo` flag. This flag sets a TTL (Time-To-Live) for caching UTXO(s) used in a transaction, preventing them from being reused in subsequent transactions within the specified time.

Reviewer: ```suggestion
To prevent this issue, you can use the `cacheUtxo` flag. This flag sets a TTL (Time-To-Live) for caching a UTXO(s) used in a transaction, preventing it from being reused in subsequent transactions within the specified time.
```

Author: @maschad My version seems to be correct already as it matches the plural "UTXO(s)" with "them."

Reviewer: So apparently there [isn't consensus](https://www.grammarphobia.com/blog/2015/04/parenthetical-plurals.html) on parenthetical plurals, I had always made the assumption they were treated as singular. So I suppose this can work.

<<< @/../../docs-snippets/src/guide/provider/provider.test.ts#options-cache-utxo{ts:line-numbers}

> It's important to note that you can only submit multiple transactions from the same account without waiting for the previous transactions to complete if your account has multiple UTXOs available. If you only have one UTXO, the first transaction will spend it, and any remaining amount will be converted into a new UTXO with a different ID.

Reviewer: I'm not fully clear on what you are trying to convey here, is my suggestion in alignment with your sentiment?

```suggestion
**Note**:

If you would like to submit multiple transactions without waiting for each transactions to complete, those transactions **must** be from the same account. If you only have one UTXO, the first transaction will spend it, and any remaining amount will be converted into a new UTXO with a different ID.
```

Author: @maschad this version has inaccurate information: `those transactions **must** be from the same account`.

Isn't the information here clear enough?

Reviewer: I'm not understanding what is being said here, are you suggesting that:

a) If I want to submit multiple transactions AND I don't want to wait on each transaction to be completed, I must have multiple UTXOs available from the same account.

b) I must submit multiple transactions from the same account without waiting for the previous transactions to be completed if I have multiple UTXOs available.

Author: @maschad Option A

Reviewer: Thanks for clarifying, I may be one of the few to misinterpret, but the wording I would suggest would be

```suggestion
**Note:**

If you would like to submit multiple transactions without waiting on each transaction to be completed, your account must have multiple UTXOs available. If you only have one UTXO, the first transaction will spend it, and any remaining amount will be converted into a new UTXO with a different ID.
```

Feel free to ignore the suggestion but the current wording isn't immediately obvious to me

_Note: If defined, `requestMiddleware`, `timeout` and `retryOptions` are applied to this custom `fetch` function as well._

<<< @/../../docs-snippets/src/guide/provider/provider.test.ts#options-fetch{ts:line-numbers}

### `cacheUtxo`

When using the SDK, it may be necessary to submit multiple transactions from the same account in a short period. In such cases, the SDK creates and funds these transactions, then submits them to the node.

However, if a second transaction is created before the first one is processed, there is a chance of using the same UTXO(s) for both transactions. This happens because the UTXO(s) used in the first transaction are still unspent until the transaction is fully processed.

> 

Reviewer 2: Might be beneficial to include a link above the [UTXO architecture](https://docs.fuel.network/docs/fuels-ts/wallets/checking-balances/#understanding-the-utxo-model) docs.

Reviewer 3: Unrelated question outside of the scope of this PR, but should this [UTXO architecture](https://docs.fuel.network/docs/fuels-ts/wallets/checking-balances/#understanding-the-utxo-model) be its own thing?

I mean, moved outside of `Wallet` > `Checking Balances`.

Author: @arboleya https://github.com/FuelLabs/fuels-ts/pull/2624

If the second transaction attempts to use the same UTXO(s) that the first transaction has already spent, it will result in the following error:

```console
Transaction is not inserted. UTXO does not exist: 0xf5...
```

This error indicates that the UTXO(s) used by the second transaction no longer exist, as the first transaction already spent them.

To prevent this issue, you can use the `cacheUtxo` flag. This flag sets a TTL (Time-To-Live) for caching UTXO(s) used in a transaction, preventing them from being reused in subsequent transactions within the specified time.

<<< @/../../docs-snippets/src/guide/provider/provider.test.ts#options-cache-utxo{ts:line-numbers}

**Note:**

> If you would like to submit multiple transactions without waiting on each transaction to be completed, your account must have multiple UTXOs available. If you only have one UTXO, the first transaction will spend it, and any remaining amount will be converted into a new UTXO with a different ID.

Reviewer 3: Small detail.

```suggestion
If you would like to submit multiple transactions without waiting for each transaction to be completed, your account must have multiple UTXOs available. If you only have one UTXO, the first transaction will spend it, and any remaining amount will be converted into a new UTXO with a different ID.
```
